simulator/src/scrypto_bindgen/translation.rs

```diff
@@ -226,18 +226,15 @@ where
         TypeKind::Custom(ScryptoCustomTypeKind::Own) => match type_validation {
             TypeValidation::Custom(ScryptoCustomTypeValidation::Own(
                 OwnValidation::IsTypedObject(package_address, bp_name),
-            )) if package_address == Some(RESOURCE_PACKAGE)
-                && bp_name == FUNGIBLE_BUCKET_BLUEPRINT =>
-            {
-                "FungibleBucket".to_owned()
-            }
-            TypeValidation::Custom(ScryptoCustomTypeValidation::Own(
-                OwnValidation::IsTypedObject(package_address, bp_name),
-            )) if package_address == Some(RESOURCE_PACKAGE)
-                && bp_name == NON_FUNGIBLE_BUCKET_BLUEPRINT =>
-            {
-                "NonFungibleBucket".to_owned()
-            }
+            )) if package_address == Some(RESOURCE_PACKAGE) => match bp_name.as_str() {
```

Comment:
I do not believe that this is right. An `OwnValidation::IsTypedObject` can never refer to buckets or proofs. Buckets and proofs have `OwnValidation::IsBucket` and `OwnValidation::IsProof` as can be seen in the `custom_well_known_types.rs` module. An `OwnValidation::IsTypedObject` can refer to an `Own<Blueprint>`. Thus, I do not believe that there is ever a case where this get hit.

Reply:
Agreed that change is good enough to merge.
However it will be refactored soon anyway.